src/nbl/asset/interchange/CGLTFLoader.cpp

```diff
@@ -754,74 +760,148 @@ namespace nbl
 
-						const auto& bufferViewID = glTFAccessor.bufferView.has_value() ? glTFAccessor.bufferView.value() : 0xdeadbeef;
-						assert(bufferViewID != 0xdeadbeef); // TODO log and nullptr
+							const uint16_t jointIDAttributeIx = cpuMeshBuffer->getJointIDAttributeIx();
 
-						const auto& glTFBufferView = glTF.bufferViews[bufferViewID];
-						const auto& bufferID = glTFBufferView.buffer.has_value() ? glTFBufferView.buffer.value() : 0xdeadbeef;
-						assert(bufferID != 0xdeadbeef);
+							for (size_t i = 0; i < cpuMeshBuffer->getIndexCount(); ++i)
+							{
+								currentJointVertexPair.vPosition = cpuMeshBuffer->getPosition(i);
+								assert(cpuMeshBuffer->getAttribute(currentJointVertexPair.vJoint, jointIDAttributeIx, i));
 
-						auto cpuBuffer = cpuBuffers[bufferID];
-						const size_t globalIBPOffset = [&]()
-						{
-							const size_t bufferViewOffset = glTFBufferView.byteOffset.has_value() ? glTFBufferView.byteOffset.value() : 0u;
-							const size_t relativeAccessorOffset = glTFAccessor.byteOffset.has_value() ? glTFAccessor.byteOffset.value() : 0u;
+								/*
+								*	TODO
+								* 
+									how to do this correctly with respect to all per-vertex joints?
+									right now it makes any sense only if we have one joint ID
+									per vertex or all joints per vertex are the same!
 
-							return bufferViewOffset + relativeAccessorOffset;
-						}();
+									got confused
+								*/
 
-						inverseBindPoseBufferBinding.buffer = core::smart_refctd_ptr(cpuBuffer);
-						inverseBindPoseBufferBinding.offset = globalIBPOffset;
+								assert(currentJointVertexPair.vJoint.x == currentJointVertexPair.vJoint.y == currentJointVertexPair.vJoint.z == currentJointVertexPair.vJoint.w);
+								const auto lower = std::lower_bound(std::begin(glTFjointNodeIDs), std::end(glTFjointNodeIDs), currentJointVertexPair.vJoint.x);
+								assert(lower != std::end(glTFjointNodeIDs));
 
+								const auto localJointNodeID = std::distance(glTFjointNodeIDs.begin(), lower); //! the ID is in range [0u, glTFjointNodeIDs.size() - 1u]
+								{
+									uint8_t* data = reinterpret_cast<uint8_t*>(inverseBindPoseBufferBinding.buffer->getPointer()) + inverseBindPoseBufferBinding.offset;
+									core::matrix4SIMD* IBPMatrix = reinterpret_cast<core::matrix4SIMD*>(data) + localJointNodeID;
 
-					}
+									auto nbl_cpp_pseudoMul4x4with3x1 = [](const core::matrix4SIMD& matrix, const core::vectorSIMDf& vector) -> core::vectorSIMDf
+									{
+										return matrix.getRow(0) * vector.x + matrix.getRow(1) * vector.y + matrix.getRow(2) * vector.z + matrix.getRow(3) * vector.z;
+									};
 
-					/*
-						TODO: skinning, pull appropriate mesh and 
-						appropriate mesh buffer via mesh buffer mutable vector
-					*/
+									const auto newInternalPoint = nbl_cpp_pseudoMul4x4with3x1(*IBPMatrix, currentJointVertexPair.vPosition);
+									jointBboxes[localJointNodeID].addInternalPoint(newInternalPoint.getAsVector3df());
```

Comment:
this one won't work with per-vertex joint IDs greater than one, or I have no clue how to compute it correctly, because multiplying `IBPMatrix` up to even 4 times per vertex position is probably wrong, but no idea, this is the reason for temporary assert above

Reply:
not an issue anymore, done